### ISSUE TITLE
Logs row code

### ISSUE THREAD

**Issue by eltonvivot** [telematics-lab/IoD_Sim#10]
Hello, 

I'm working on a script to split the log file into smaller log files, but I didn't understand what means the second column of each row. First, I thought that it was the log corresponding to a node ID (except for -1 that seems to log scenarion configurations). This way I was specting the pattern _Column code X | Log of Node X_. 
However, in the log are rows with code 1 informing CourseChange for node 0:
`+1.020933985s 1 LteEnbNetDevice:Send(0x7f400c036230, 0x7f400c0e2d40, 02-06-ff:ff:ff:ff:ff:ff, 2048)
+1.022000000s 1 DroneClientApplication:CourseChange(): [INFO ] +1.022e+09ns [Node 0]; Pos: (150.024:260.071:0.317207); Vel: (0.0373284:0.110076:0.174899)` 

I also do have the CourseChange log with 0 as second column value, and the location values of both row codes are different:
`+1.023999999s 0 DroneClientApplication:ReceivePacket(): [INFO ] [Node 0] client received 26 bytes from 200.0.0.1
+1.023999999s 0 DroneClientApplication:ReceivePacket(): [INFO ] [Node 0] packet contents: {"cmd":"HELLO_ACK","sn":0}
+1.023999999s 0 DroneClientApplication:ReceivePacket(): [INFO ] [Node 0] received HELLO_ACK with IP 200.0.0.1
+1.025000000s 0 DroneClientApplication:CourseChange(): [INFO ] +1.025e+09ns [Node 0]; Pos: (150.029:260.085:0.336894); Vel: (0.0452322:0.133603:0.194176)
+1.027000000s 0 DroneClientApplication:CourseChange(): [INFO ] +1.027e+09ns [Node 0]; Pos: (150.034:260.1:0.356572); Vel: (0.0518335:0.15286:0.203831)`

Do you know how the value of the second column is assigned to each row?

Best,
Elton.

**Comment by GiovanniGrieco** [comment]
Hi Elton,

At first glance, the macro `NS_LOG_INFO` references the `log-macros-enabled.h` file in ns-3, which gives the following message template:
```
#define NS_LOG(level, msg)                                                                         \
    NS_LOG_CONDITION                                                                               \
    do                                                                                             \
    {                                                                                              \
        if (g_log.IsEnabled(level))                                                                \
        {                                                                                          \
            NS_LOG_APPEND_TIME_PREFIX;                                                             \
            NS_LOG_APPEND_NODE_PREFIX;                                                             \
            NS_LOG_APPEND_CONTEXT;                                                                 \
            NS_LOG_APPEND_FUNC_PREFIX;                                                             \
            NS_LOG_APPEND_LEVEL_PREFIX(level);                                                     \
            std::clog << msg << std::endl;                                                         \
        }                                                                                          \
    } while (false)
```
Surely, this means that the second column of the log file is the `ns3::Node` Identifier that is executing such action.
I'd like to reproduce this behavior. What JSON file should I execute? Can you also give me indication of the commit that you are working on (pick it from `git log`)?

Thanks

**Comment by GiovanniGrieco** [comment]
Hi Elton,

I debugged the code and found the following observations. First of all, `NS_LOG_APPEND_NODE_PREFIX` expands to `Simulator::GetContext()`, which is a static call to retrieve the current Node Identifier in the simulation context, i.e., the event to handle at that time instant. In your scenario, this context can vary between Node ID `0` and `1`, depending whose event we are processing, drone or ZSP, respectively.

Lots of events are Tx/Rx of a frame. For instance, if we have to handle the event of a frame being transmitted by the ZSP (so Node ID and context equal `1`), the simulator will query the channel model, which executes the code in `ns3::MultiModelSpectrumChannel`:
```
Angles txAngles(receiverMobility->GetPosition(), txMobility->GetPosition());
txAntennaGain = rxParams->txAntenna->GetGainDb(txAngles);
NS_LOG_LOGIC("txAntennaGain = " << txAntennaGain << " dB");
pathLossDb -= txAntennaGain;
```

As you can see, the channel model requests the current position of both the transmitter and receiver nodes, so in our case the transmitter is the ZSP and the receiver is, supposedly, the Drone. This is needed to evaluate the instantaneous channel loss.
Why instantaneous? I think that this is an approximation, given that the electromagnetic wave is travelling at the speed of light. 

Consequently, the instruction `receiverMobility->GetPosition()` queries the current position of the drone, which is a call that is handled by `ns3::ParametricSpeedDroneMobilityModel` -as configured in the given JSON file. Given such position, if it differs from past ones, the model triggers a `CourseChange` trace event, which is propagated to every event handler listening to this event. This includes the `ns3::DroneClientApplication` of the Drone with Node ID `0`, which happens to have a `LOG_INFO` instruction to debug these new coordinates.

In short: this log message with Context/Node ID `1` reporting a `CourseChange` trace event of a Node with ID `0` is by design. If you don't like this behavior, feel free to change the application logic!
